pyproject.toml

```
@@ -1,6 +1,6 @@
 [build-system]
 # Setuptools version should match setup.py; wheel because pip will insert it noisily
-requires = ["setuptools >= 42.0.0", "versioningit ~= 0.3.0", "wheel"]
+requires = ["setuptools >= 42.0.0", "versioningit", "wheel"]
```

shouldn't it be 
```suggestion
requires = ["setuptools >= 42.0.0", "versioningit >= 0.3.0", "wheel"]
```
then so some pip env with outdated version would not just silently proceed?

---

Projects with `[build-system]` tables should always be built in isolated environments with freshly-installed dependencies (e.g., via [build](https://github.com/pypa/build)).  A "pip env with an outdated version" should have no effect.

---

then we should not have versioning limit on `setuptools` either (consistency is good) and why did we get it there in the first place?

---

`>= 42.0.0` just sets a lower bound (as opposed to `~= 0.3.0`, which also sets an upper bound), so it's harmless.

> why did we get it there in the first place?

Copy & paste from an earlier era.  When setuptools v42.0.0 was new, I believe it was possible for isolated build environments to be created with an older setuptools version automatically installed (because `venv` and `virtualenv` both bundle setuptools instead of downloading the latest version every time an environment is created), so specifying a minimum setuptools version forced an update if the environment's version was older.